### PR TITLE
8331077 : nroff man page update for jar tool

### PR DESCRIPTION
**Pull request by weibxiao**
nroff man page update for jar tool. <br>
This update is caused by the change of https://bugs.openjdk.org/browse/JDK-8318971. While the .md man pages got updated in other repos, the corresponding nroff man page was never updated in OpenJDK repos

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331077](https://bugs.openjdk.org/browse/JDK-8331077): nroff man page update for jar tool (**Sub-task** - P4)

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)
 * [Sean Coffey](https://openjdk.org/census#coffeys) (@coffeys - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19039/head:pull/19039` \
`$ git checkout pull/19039`

Update a local copy of the PR: \
`$ git checkout pull/19039` \
`$ git pull https://git.openjdk.org/jdk.git pull/19039/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19039`

View PR using the GUI difftool: \
`$ git pr show -t 19039`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19039.diff">https://git.openjdk.org/jdk/pull/19039.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19039#issuecomment-2088845844)

### CODE REVIEW

File: src/jdk.jartool/share/man/jar.1 [hunk 1]
```diff
@@ -35,7 +35,7 @@
 . ftr VB CB
 . ftr VBI CBI
 .\}
-.TH "JAR" "1" "2024" "JDK 23-ea" "JDK Commands"
+.TH "JAR" "1" "2024" "JDK 23-internal" "JDK Commands"
```
Comment (jonathan-gibbons) [inline src/jdk.jartool/share/man/jar.1:38]: should remain as `ea` -- not `internal`

**Review by jonathan-gibbons — CHANGES_REQUESTED** [review]

**Comment by weibxiao** [comment]
reverted to "JDK 23-ea"

**Review by coffeys — APPROVED** [review]
Looks ok to me

**Review by jonathan-gibbons — APPROVED** [review]
Approved, as corresponding to the upstream Markdown file.

That being said, the descriptions about arg files in lines 334-343 look somewhat clunky, and could benefit from some improvements in a later update.

**Pull request closed without merging**